R/clustering.R

```diff
@@ -676,79 +687,35 @@ RunLeiden <- function(
   weights = NULL,
   node.sizes = NULL,
   resolution.parameter = 1,
-  random.seed = 0
+  random.seed = 0,
+  n.iter = 10
 ) {
   if (!py_module_available(module = 'leidenalg')) {
     stop("Cannot find Leiden algorithm, please install through pip (e.g. pip install leidenalg).")
   }
-  # import python modules with reticulate
-  leidenalg <- import(module = "leidenalg")
-  ig <- import(module = "igraph")
-  # convert matrix input
-  adj_mat <- as.matrix(x = ceiling(x = adj_mat))
-  # convert to python numpy.ndarray, then a list
-  adj_mat_py <- r_to_py(x = adj_mat)
-  adj_mat_py <- adj_mat_py$tolist()
-  # convert graph structure to a Python compatible object
-  snn_graph <- ig$Graph$Adjacency(adj_mat_py)
-  # compute partitions
-  part <- switch(
-    EXPR = partition.type,
-    'RBConfigurationVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$RBConfigurationVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      resolution_parameter = resolution.parameter,
-      seed = as.integer(x = random.seed)
-    ),
-    'ModularityVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$ModularityVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      seed = as.integer(x = random.seed)
-    ),
-    'RBERVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$RBERVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      node_sizes = node.sizes,
-      resolution_parameter = resolution.parameter,
-      seed = as.integer(x = random.seed)
-    ),
-    'CPMVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$CPMVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      node_sizes = node.sizes,
-      resolution_parameter = resolution.parameter,
-      seed = as.integer(x = random.seed)
-    ),
-    'MutableVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$MutableVertexPartition,
-      initial_membership = initial.membership
-    ),
-    'SignificanceVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$SignificanceVertexPartition,
-      initial_membership = initial.membership,
-      node_sizes = node.sizes,
-      resolution_parameter = resolution.parameter
-    ),
-    'SurpriseVertexPartition' = leidenalg$find_partition(
-      snn_graph,
-      leidenalg$SurpriseVertexPartition,
-      initial_membership = initial.membership,
-      weights = weights,
-      node_sizes = node.sizes
-    ),
-    stop("please specify a partition type as a string out of those documented")
+  if (method == "matrix") {
+    #cast to dense (supported by reticulate for numpy.array)
```

Comment:
`input` is never used, should this be `object`?

Reply:
I've updated the `input` to be the object passed to `leiden::leiden`. This is purely to avoid ambiguity which `object` is being passed to python.